syft/serde/serde.py

```diff
@@ -124,19 +125,40 @@ def _force_full_simplify(obj: object) -> object:
     Returns:
         The simplified object
     """
-    current_type = type(obj)
-
-    if current_type in forced_full_simplifiers:
-        left = forced_full_simplifiers[current_type][0]
-
-        right = forced_full_simplifiers[current_type][1]
-        right = right(obj)
-
-        result = (left, right)
-    else:
-        result = _simplify(obj)
+    try:
+        # check to see if there is a simplifier
+        # for this type. If there is, return the simplified object.
+        current_type = type(obj)
+        result = (
+            forced_full_simplifiers[current_type][0],
+            forced_full_simplifiers[current_type][1](obj),
+        )
+        return result
+    except KeyError:
+        # If the object type is not in forced_full_simplifiers,
+        # we check the classes that this object inherits from.
+        # `inspect.getmro` give us all types this object inherits
+        # from, including `type(obj)`. We can skip the type of the
+        # object because we already tried this in the
+        # previous step.
+        classes_inheritance = inspect.getmro(type(obj))[1:]
+
+        for inheritance_type in classes_inheritance:
+            if inheritance_type in forced_full_simplifiers:
+                # Store the inheritance_type in forced_full_simplifiers so next
+                # time we see this type serde will be faster.
+                forced_full_simplifiers[current_type] = forced_full_simplifiers[inheritance_type]
+                result = (
+                    forced_full_simplifiers[current_type][0],
+                    forced_full_simplifiers[current_type][1](obj),
+                )
+                return result
 
```

Comment:
Also add a registry for the types where nothing was found (like int, float etc) otherwise you'll end up always doing the inheritance check for these. It can be a simple set object let's say no_simplifiers_found = {}
```python
    except KeyError:
        type_obj = type(obj)
        if type_obj  not in no_simplifiers_found:
            classes_inheritance = inspect.getmro(type_obj)[1:]
            for inheritance_type in classes_inheritance:
                if inheritance_type in forced_full_simplifiers:
                    # Store the inheritance_type in forced_full_simplifiers so next
                    # time we see this type serde will be faster.
                    forced_full_simplifiers[current_type] = forced_full_simplifiers[inheritance_type]
                    result = (
                    forced_full_simplifiers[current_type][0],
                    forced_full_simplifiers[current_type][1](obj),
                    )
                    return result
             no_simplifiers_found.add(type_obj)
       else:
            return _simplify(obj)
```